src/Uplink/Admin/Provider.php

```diff
@@ -181,14 +181,22 @@ public function filter_upgrader_install_package_result( $result, $extras ) {
 	 *
 	 * @since 1.0.0
 	 *
-	 * @param string       $source        File source location.
-	 * @param mixed        $remote_source Remote file source location.
-	 * @param \WP_Upgrader $upgrader      WP_Upgrader instance.
-	 * @param array        $extras        Extra arguments passed to hooked filters.
+	 * @param string|\WP_Error $source        File source location or a WP_Error.
+	 * @param mixed            $remote_source Remote file source location.
+	 * @param \WP_Upgrader     $upgrader      WP_Upgrader instance.
+	 * @param array            $extras        Extra arguments passed to hooked filters.
 	 *
 	 * @return string|\WP_Error
 	 */
 	public function filter_upgrader_source_selection_for_update_prevention( $source, $remote_source, $upgrader, $extras ) {
+		if ( is_wp_error( $source ) ) {
+			return new \WP_Error(
+				'stellarwp-uplink-updater-failed-prevention',
+				esc_html__( 'Your update failed due to an unknown error.', '%TEXTDOMAIN%' ),
+				[]
+			);
+		}
```

Comment:
Maybe we just return the WP_Error if we have it and not change the error message:

```suggestion
		if ( is_wp_error( $source ) ) {
			return source;
		}
```